addons/l10n_pk_edi/models/account_move.py

```diff
@@ -0,0 +1,494 @@
+
+    def _l10n_pk_validate_moves(self):
+        corrupt_move_name = {}
+        move_ids = [move for move in self if move.is_sale_document()]
+        for move in move_ids:
+            if err_msg := move._l10n_pk_validate_move_lines():
+                corrupt_move_name[move.name] = ('\n').join(err_msg)
+        if corrupt_move_name:
+            error_msg = "\n\n".join(f"{name}:\n{err_msg}" for name, err_msg in corrupt_move_name.items())
+            return {
+                "title": "Please configure following details to confirm Invoices.",
+                "msg": error_msg,
+            }
+        for move in move_ids:
+            if (move.move_type == 'out_refund'
+                and (not move.l10n_pk_edi_cancel_reason or not move.l10n_pk_edi_cancel_remarks)
+            ):
+                corrupt_move_name[move.name] = 'error'
+        if corrupt_move_name:
+            error_msg = "\n".join(f"- {name}" for name, msg in corrupt_move_name.items())
+            return {
+                "title": "To confirm credit/debit note, set cancel reason and remarks "
+                         "at Electronic Invoicing tab in following invoices:",
+                "msg": error_msg,
+            }
+        return False
+
+    def _l10n_pk_validate_move_lines(self):
+        error_messages = []
+        for line in self.invoice_line_ids.filtered(
+            lambda line: line.display_type == 'product'
+            and not self._l10n_pk_is_global_discount(line)
+        ):
+            if line.discount < 0:
+                error_messages.append(_("- Negative discount is not allowed, set in line %s", line.name))
+            pct_code = self._l10n_pk_edi_extract_digits(line.product_id.l10n_pk_edi_pct_code)
+            if not pct_code:
+                error_messages.append(_("- PCT code is not set in product line %s", line.name))
+            elif not re.match("^[a-zA-Z0-9]{8}$", pct_code):
+                error_messages.append(_(
+                    "- The PCT Code '%(pct_code)s' in product line %(product_line)s is invalid."
+                    " Please ensure the code contains exactly 8 digits.",
+                        pct_code=pct_code,
+                        product_line=line.product_id.name or line.name
+                    )
+                )
+        return error_messages
+
+    def _get_l10n_pk_edi_payment_mode(self):
+        """
+            Payment Mode for invocie according to FBR:
+            1. Cash
+            2. Card
+            3. Gift Voucher
+            4. Loyalty Card
+            5. Mixed
+            6. Cheque
+        """
+        payments = self.pos_order_ids.mapped('payment_ids')
+        if len(payments) > 1:
+            # If multiple payment modes are used,
+            # Return 5 to represent 'Mixed' as the payment mode.
+            return 5
+        # 1 represents the default 'Cash' payment mode.
+        return 2 if payments[0].payment_method_id.type in ['bank', 'card'] else 1
```

Comment:
:\ where is the use of `3`? can we be specific for the cash?

Reply:
FBR provides various payment options. however, our POS system does not support certain methods such as `Gift Vouchers`, `Loyalty Cards`, and `Cheques`. As a result, I have not included cases for these payment types.